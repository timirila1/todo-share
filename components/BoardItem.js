import React from 'react';
import {
    TouchableHighlight,
    Text,
    StyleSheet,
    View,
    Animated,
} from 'react-native';
import PropTypes from 'prop-types';

import { colors } from '../services/consts';

export const BoardItem = (props) => {
    const {
        board,
        onPress,
        onLongPress,
        blocked,
        selected,
    } = props;

    const borderAppearing = new Animated.Value(0);
    const opacityAnimation = new Animated.Value(1);

    Animated
        .timing(borderAppearing, { toValue: selected ? 4 : 0, duration: 50 })
        .start();

    Animated
        .timing(opacityAnimation, { toValue: blocked ? 0.7 : 1, duration: 50 })
        .start();

    return (
        <View style={styles.wrapper}>
            <TouchableHighlight
                style={styles.underlay}
                underlayColor="black"
                onPress={onPress}
                onLongPress={onLongPress}
            >
                <Animated.View style={[
                    styles.item,
                    selected ? styles.selected : {},
                    { opacity: opacityAnimation },
                    { borderWidth: borderAppearing },
                ]}>
                    <Text style={styles.title}>{board.title}</Text>
                    <Text style={styles.desc}>{board.desc}</Text>
                </Animated.View>
            </TouchableHighlight>
        </View>
    );
};

const styles = StyleSheet.create({
    wrapper: {
        width: '50%',
        paddingRight: 20,
        marginBottom: 20,
    },
    underlay: {
        minHeight: 250,
        borderRadius: 20,
    },
    item: {
        flex: 1,
        backgroundColor: colors.main,
        padding: 20,
        borderRadius: 20,
    },
    selected: {
        borderColor: colors.additional,
    },
    title: {
        color: 'white',
        fontSize: 24,
        fontWeight: '300',
        paddingBottom: 10,
    },
    desc: {
        fontWeight: '200',
        color: colors.textLight,
    },
});
