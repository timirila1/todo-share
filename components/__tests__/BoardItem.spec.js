import 'react-native';
import React from 'react';
import { BoardItem } from '../BoardItem';
import renderer from 'react-test-renderer';

it('Render BoardsItem', async () => {
    const boardMock = {
        id: 1,
        title: 'Test title',
        desc: 'Test desc'
    };

    const tree = renderer.create(
        <BoardItem
            index={1}
            navigation={{}}
            board={boardMock}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
