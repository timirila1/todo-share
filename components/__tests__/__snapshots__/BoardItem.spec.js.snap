// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BoardsItem 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
      "paddingRight": 20,
      "width": "50%",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 20,
        "minHeight": 250,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#5e1c73",
            "borderRadius": 20,
            "flex": 1,
            "padding": 20,
          },
          Object {},
          Object {},
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 24,
            "fontWeight": "300",
            "paddingBottom": 10,
          }
        }
      >
        Test title
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.8)",
            "fontWeight": "200",
          }
        }
      >
        Test desc
      </Text>
    </View>
  </View>
</View>
`;
